Spring Boot Practice. Order 1.0

aksoli666 opened the pull request:
added Order and OrderItem models, added repository, service and controller layers;

aksoli666 committed:
Spring Boot Practice. Order 1.0
added Order and OrderItem models, added repository, service and controller layers;

Elena-Bruyako commented on src/main/java/shop/entity/Order.java:
```diff
@@ -0,0 +1,54 @@
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+    @OneToOne
+    @JoinColumn(name = "status_id", nullable = false)
```
```suggestion
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
```

Elena-Bruyako commented on src/main/java/shop/entity/Status.java:
```diff
@@ -0,0 +1,31 @@
+@Table(name = "statuses")
+@Getter
+@Setter
+public class Status {
```
remove this entity, create Status enum in Order entity

Elena-Bruyako commented on src/main/java/shop/entity/User.java:
```diff
@@ -81,4 +82,9 @@ public boolean isCredentialsNonExpired() {
         return !isDeleted;
     }
+
+    public static Long getUserIdFromAuthentication(Authentication authentication) {
```
I don't think that this method is appropriate in User entity

Elena-Bruyako commented on src/main/java/shop/service/impl/OrderItemServiceImpl.java:
```diff
@@ -0,0 +1,45 @@
+
+@Service
+@RequiredArgsConstructor
+public class OrderItemServiceImpl implements OrderItemService {
```
all logic should be in OrderService

Elena-Bruyako commented on src/main/java/shop/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,104 @@
+    @Override
+    public OrderResponseDto placeOrder(Authentication authentication,
+                                       PlaceOrderRequestDto requestDto) {
+        User user = (User)userDetailsService.loadUserByUsername(authentication.getName());
```
```suggestion
        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
```

Elena-Bruyako commented on src/main/java/shop/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,104 @@
+
+    @Transactional
+    @Override
+    public OrderResponseDto placeOrder(Authentication authentication,
```
method is too big, let's divide logic to separate methods

Elena-Bruyako commented on src/main/resources/db/changelog/changes/11-create-statuses-table.yaml:
```diff
@@ -0,0 +1,20 @@
+databaseChangeLog:
+  - changeSet:
```
remove

Elena-Bruyako reviewed (changes requested):
implementation should be improved, then let's check again

aksoli666 committed:
Spring Boot Practice. Order 2.0
replaced logic OrderItemService(Controller) to OrderService
deleted Status entity and replace enum Status to Order entity
divided logic to separate methods -> placeOrder(...)

Elena-Bruyako commented on src/main/java/shop/controller/BookController.java:
```diff
@@ -37,7 +37,7 @@ public class BookController {
             description = "Create a new book"
     )
-    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    //@PreAuthorize("hasRole('ROLE_ADMIN')")
```
remove commented code everywhere

Elena-Bruyako commented on src/main/java/shop/controller/CategoryController.java:
```diff
@@ -31,7 +31,7 @@ public class CategoryController {
             summary = "Create a new category",
             description = "Create a new category")
-    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    //@PreAuthorize("hasRole('ROLE_ADMIN')")
```
same here

Elena-Bruyako commented on src/main/java/shop/controller/OrderController.java:
```diff
@@ -0,0 +1,88 @@
+            summary = "Update order status",
+            description = "Update order status"
+    )
+    //@PreAuthorize("hasRole('ROLE_ADMIN')")
```
```suggestion
    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Elena-Bruyako commented on src/main/java/shop/mapper/OrderMapper.java:
```diff
@@ -0,0 +1,19 @@
+import shop.dto.responce.order.OrderResponseDto;
+import shop.entity.Order;
+
+@Mapper(config = MapperConfig.class, uses = {OrderItemMapper.class})
```
```suggestion
@Mapper(config = MapperConfig.class, uses = OrderItemMapper.class)
```

Elena-Bruyako commented on src/main/java/shop/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,137 @@
+    private Order createOrder(User user, PlaceOrderRequestDto requestDto) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.COMPLETED);
```
I think when you just create an order status should be Pending

Elena-Bruyako commented on src/main/resources/db/changelog/changes/13-create-orders-items-table.yaml:
```diff
@@ -0,0 +1,26 @@
+      author: dorofieieva
+      changes:
+        - createTable:
+            tableName: orders_items
```
remove this table

Elena-Bruyako reviewed (changes requested):
see comments

aksoli666 committed:
Spring Boot Practice. Order 3.0
fixed:
